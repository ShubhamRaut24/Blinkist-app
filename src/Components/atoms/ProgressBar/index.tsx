import { LinearProgress } from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';
import { Theme } from "../../../Themes/theme";

const barStyled =
{ height: '20px', 
  opacity: '0.5', 
  borderRadius: '0px 0px 10px 10px', 
  backgroundColor: '#DFE8F6'
}

const ProgressBarComponent = ({theme, ...props}:any) => {
    return (
        <ThemeProvider theme={Theme}>
            <LinearProgress {...props} sx={barStyled} variant="determinate" />
        </ThemeProvider>
    )
}

export default ProgressBarComponent;