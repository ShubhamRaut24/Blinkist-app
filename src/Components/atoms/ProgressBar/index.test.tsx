import React from 'react';
import ProgressBar from ".";

import renderer from 'react-test-renderer';

const barStyled =
{ height: '15px', 
  opacity: '0.5', 
  borderRadius: '0px 0px 10px 10px', 
  backgroundColor: '#DFE8F6'
}

test('ProgressBar', () => {
    const component = renderer.create(
   
            <ProgressBar value={60} sx={barStyled} variant="determinate"></ProgressBar>
       
    );
    expect(component).toMatchSnapshot();
});