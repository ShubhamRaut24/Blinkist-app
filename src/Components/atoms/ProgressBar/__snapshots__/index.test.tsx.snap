// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar 1`] = `
<span
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={60}
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-dp0sbv-MuiLinearProgress-root"
  role="progressbar"
>
  <span
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
    style={
      Object {
        "transform": "translateX(-40%)",
      }
    }
  />
</span>
`;
